### ISSUE TITLE
Setup regular GitHub repo syncing for users?

### ISSUE THREAD

**Issue by roll** [frictionlessdata/goodtables.io#91]
# Overview

For now it could be done only on-demand (clicking on `Sync repos` button).

# Thoughts

- we need to store in database github tokens
- we need to setup Etags approach (Etags for every repo/user should be stored somewhere)

Not sure it's priority for now because on-demand syncing seems enough convenient on this stage.

**Comment by pwalsh** [comment]
Isn't this just an instance of https://github.com/frictionlessdata/goodtables.io/issues/89 ?

The pattern here is well established in travis and coveralls.

**Comment by roll** [comment]
@pwalsh 
I would say this issue is concrete feature (add regular syncing). And #89 is low-level issue with storing celery task ids. 

May be we should start to distinct things by feature/enhancement (cc @amercader )

**Comment by pwalsh** [comment]
@roll is this not the only instance of the pattern needed for #89 ? in which case, I'd merge the two.

**Comment by roll** [comment]
@pwalsh 
pretty sure #89 will cover much more things (for every slow query we should create celery task)

But I suppose we could merge them and as a part of adding regular syncing to add task - `setup database table to handle celery tasks not related to jobs etc`)

**Comment by pwalsh** [comment]
ok. @amercader I suggest merging this and #89 and also think relates to #92 somewhat. But up to you when you do your estimates for the milestone.

**Comment by roll** [comment]
@amercader 
This issue seems out-of-date. 

What we have:
- all frontend work for github syncing is done

What we're left:
- updates happen only if USER CALL THE SYNC button. There is no auto background updates (I suppose it was initial goal of this issue)

**Comment by pwalsh** [comment]
@amercader @roll Travis and Coveralls only sync repos on user action - it should be good enough for us too.

**Comment by roll** [comment]
@pwalsh 
@amercader 
I'm pretty agree on it. At least for first year of service work. Or before user request. I always sync my travis repos by myself and don't have any problems with honestly..

**Comment by pwalsh** [comment]
yes - I'd say automated repo syncing is clearly a non goal.

**Comment by amercader** [comment]
Fair enough, regular syncing was just mentioned as potential option but happy to park it for now. And for the sake of completeness Travis syncs every time you log in :)

I still think that using Etags / Last-Modified Header for GH API calls specially when syncing is well worth doing so I'll create a backlog issue for it.
